c2corg_api/search/sync.py

```diff
@@ -0,0 +1,56 @@
+import transaction
+import logging
+
+from c2corg_api.search import elasticsearch_config
+from c2corg_api.search.mapping import SearchDocument
+from c2corg_api.search.utils import strip_bbcodes
+
+log = logging.getLogger(__name__)
+
+
+def run_on_successful_transaction(operation):
+    """Run the given operation when the current transaction is committed
+    successfully.
+    """
+    def run_when_successful(success, *args, **kws):
+        if success:
+            try:
+                operation()
+            except Exception as exc:
+                log.error('Scheduled operation failed', exc)
+        else:
+            log.info('Scheduled operation is not run because transaction '
+                     'was not successful')
+
+    current_transaction = transaction.get()
+    current_transaction.addAfterCommitHook(run_when_successful)
+
+
+def sync_search_index(document):
+    """Update the document in ElasticSearch. If the document is not yet present
+     in the ElasticSearch index, it will be created.
+     The operation will be run once the current transaction has been committed
+     successfully.
+    """
+    document_id = document.document_id
+    doc = {
+        'doc_type': document.type
+    }
+    for locale in document.locales:
```

Comment:
You mean constructing the document for ES? At this moment I have all the data I need. When the transaction is closed, the db session is also closed (?) and the data might have to be loaded again (not sure about this).

This will change with #89 anyway.